akka-cluster-sharding/src/main/scala/akka/cluster/sharding/Shard.scala

```diff
@@ -359,8 +359,9 @@ private[akka] class Shard(
     case None =>
       log.debug("HandOff shard [{}]", shardId)
 
-      if (state.entities.nonEmpty) {
+      if (idByRef.nonEmpty) {
         val entityHandOffTimeout = (settings.tuningParameters.handOffTimeout - 5.seconds).max(1.seconds)
+        log.debug("Starting HandOffStopper for shard {} to terminate following entities: {}", shardId, idByRef.keySet)
```

Comment:
I think that could risk being a very large number of ids, maybe log the number of entities instead to not do a denial of service on the logfile?

Reply:
good point, I've changed to log number of entities